Common/src/geometry/dual_grid/CPoint.cpp

```diff
@@ -26,430 +26,139 @@
  */
 
 #include "../../../include/geometry/dual_grid/CPoint.hpp"
+#include "../../../include/CConfig.hpp"
+#include "../../../include/omp_structure.hpp"
 
-CPoint::CPoint(unsigned short val_nDim, unsigned long val_globalindex, CConfig *config) : CDualGrid(val_nDim) {
 
-  unsigned short iDim, jDim;
+CPoint::CPoint(unsigned long npoint, unsigned long ndim, const CConfig *config) : nDim(ndim) {
 
```

Comment:
We might be able to save some memory by being a bit more selective in what gets allocated in this constructor, if people have thoughts let me know (one obvious one would be all the MG stuff).